### PR TITLE
Replaced 'Deploy-Default-Udr' policy with 'Deploy-Custom-Route-Table'

### PR DESCRIPTION
**Pull request by fguerri**
## Overview/Summary

This PR replaces the 'Deploy-Default-Udr' policy with a new one named 'Deploy-Custom-Route-Table' that allows more granular control on which UDRs should exist in compliant route tables associated to subnets in spoke VNets in hub&spoke topologies.

## This PR fixes/adds/changes/removes

1. Removes the policy definition 'Deploy-Default-Udr' from policies.json
2. Adds a new policy definition 'Deploy-Custom-Route-Table' to policies.json
3. Updates ESLZ-Policies.md to document the new policy
4. Updates Whats-new.md

### Breaking Changes

No breaking changes. The new 'Deploy-Custom-Route-Table' policy implements the same logic as the 'Deploy-Default-Udr' policy. When a compliant route table cannot be found in a VNet's resource group, a route table is deployed which contains all the UDRs provided as an input parameter when assigning the policy. The policy does NOT automatically associate the complaint route table to subnets in scope.

## Testing Evidence

1. The ESLZ solution accelerator was successfully deployed after committing the changes listed above. 
2. The new policy definition is available and the policy can be assigned.
3. The policy works as expected for new VNets that are deployed within the assignment scope. A compliant custom route table is deployed if it not exists in the VNet's resource group.
4. Pre-existing VNets are marked as "Not compliant" and can be remediated with with a Remediation Task. After running the Remediation Task, VNets are marked as "Compliant".

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [ ] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.
- [x] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

File: eslzArm/managementGroupTemplates/policyDefinitions/policies.json [hunk 1]
```diff
@@ -879,41 +879,55 @@
...
                 {
                     "properties": {
-                        "displayName": "Deploy a user-defined route to a VNET with specific routes.",
-                        "policyType": "Custom",
-                        "mode": "Indexed",
-                        "description": "Deploy a user-defined route to a VNET with routes from spoke to hub firewall. This policy must be assigned for each region you plan to use.",
+                        "displayName": "Deploy a route table with specific user defined routes",
+                        "description": "Deploys a route table with specific user defined routes when one does not exist. The route table deployed by the policy must be manually associated to subnet(s)",
                         "metadata": {
                             "version": "1.0.0",
```
Comment (krnese) [inline eslzArm/managementGroupTemplates/policyDefinitions/policies.json:885]: Should we bump the version, to 1.1.0?

File: eslzArm/managementGroupTemplates/policyDefinitions/policies.json [hunk 2]
```diff
@@ -933,75 +947,139 @@
...
+                                                }
                                             }
                                         }
                                     }
                                 }
                             }
                         }
                     },
-                    "name": "Deploy-Default-Udr"
+                    "name": "Deploy-Custom-Route-Table"
```
Comment (krnese) [inline eslzArm/managementGroupTemplates/policyDefinitions/policies.json:1082]: Is it intentional to give it a new name? That will result in a net-new policyDefinition from an ARM perspective. @krowlandson - thoughts on this from TF side?

**Review by krnese — CHANGES_REQUESTED** [review]

File: eslzArm/managementGroupTemplates/policyDefinitions/policies.json (hunk 2 above)
Comment (fguerri) [reply eslzArm/managementGroupTemplates/policyDefinitions/policies.json:1082]: To @krowlandson's point, the intent was to implement a new policy that allows defining all the aspects of a spoke VNet's routing policy (UDRs and BGP propagation settings). Hence the new name and the "1.0.0" version. If I am missing anything that makes it difficult/not desirable to replace the existing policy, I am happy to prepare a PR that updates it, keeps the name and changes version to "1.1.0"

Comment (krnese) [reply eslzArm/managementGroupTemplates/policyDefinitions/policies.json:1082]: Sounds good to me, replacing the one we had with a new, and keeping it as 1.0.0.

**Review by krnese — APPROVED** [review]

**Pull request merged**